external/fv3fit/tests/emulation/thermobasis/test_models.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import pytest
+import tensorflow as tf
+from fv3fit.emulation.thermobasis.loss import QVLossSingleLevel, RHLossSingleLevel
+from fv3fit.emulation.thermobasis.models import (
+    RHScalarMLP,
+    ScalarMLP,
+    UVTQSimple,
+    UVTRHSimple,
+    V1QCModel,
+)
+from fv3fit.emulation.thermobasis.thermo import RelativeHumidityBasis
+from fv3fit.emulation.thermobasis.emulator import (
+    get_model,
+    Config as OnlineEmulatorConfig,
+)
+
+from utils import _get_argsin
+
+
+@pytest.mark.parametrize("with_scalars", [True, False])
+def test_V1QCModel(with_scalars):
+    x = _get_argsin(3)
+
+    if with_scalars:
+        n_scalars = 2
+        x.scalars = [tf.random.uniform([x.q.shape[0], n_scalars])]
+    else:
+        n_scalars = 0
+
+    model = V1QCModel(x.u.shape[1])
+    assert not model.scalers_fitted
+    model.fit_scalers(x, x)
+    assert model.scalers_fitted
+    y = model(x)
+    assert isinstance(y, RelativeHumidityBasis)
+
+
+def test_UVTQSimple():
+    model = UVTQSimple(10, 10, 10, 10)
+    shape = (3, 10)
+    argsin = _get_argsin(levels=10, n=3)
+    model.fit_scalers(argsin, argsin)
+    out = model(argsin)
+
+    assert tuple(out.u.shape) == shape
+    assert tuple(out.v.shape) == shape
+    assert tuple(out.T.shape) == shape
+    assert tuple(out.q.shape) == shape
+
+
+def test_tf_dataset_behaves_as_expected_for_tuples():
+    u = tf.ones((10, 1))
+    d = tf.data.Dataset.from_tensor_slices(((u, u), (u, u)))
+    (out, _), (_, _) = next(iter(d))
+    assert (1,) == tuple(out.shape)
```

Comment:
Yah. Just figuring out how tensorflow works. The clean code books recommends turning these kinds of experiments into tests rather than just leaving them as ipython exploration.